src/plugins/data/server/search/session/session_service.ts

```diff
@@ -38,24 +50,129 @@ export type ISearchSessionClient = ReturnType<
   ReturnType<BackgroundSessionService['asScopedProvider']>
 >;
 
+export interface SessionInfo {
+  insertTime: Moment;
+  retryCount: number;
+  ids: Map<string, string>;
+}
+
 export class BackgroundSessionService {
   /**
    * Map of sessionId to { [requestHash]: searchId }
    * @private
    */
-  private sessionSearchMap = new Map<string, Map<string, string>>();
+  private sessionSearchMap = new Map<string, SessionInfo>();
+  private internalSavedObjectsClient!: SavedObjectsClientContract;
+  private monitorInterval!: NodeJS.Timeout;
+
+  constructor(private readonly logger: Logger) {}
 
-  constructor() {}
+  private setupMonitoring = (savedObjects: SavedObjectsServiceStart) => {
+    // TODO: setup monitoring only if BGS is enabled
+    const internalRepo = savedObjects.createInternalRepository();
+    this.internalSavedObjectsClient = new SavedObjectsClient(internalRepo);
```

Comment:
It looks like the `BACKGROUND_SESSION_TYPE` saved object type is space-aware, but creating a saved objects client this way does not create a space-aware client as far as I'm aware. What happens to sessions created outside of the default space? I expect that this service won't be able to interact with them properly

Reply:
If an object is created outside of the default space, can an internal repository initiated this way update them/delete them? They should always be created from a user-initiated request, but they won't always be updated in the same manner.